grails-app/services/org/pih/warehouse/core/OrganizationIdentifierService.groovy

```diff
@@ -0,0 +1,75 @@
+package org.pih.warehouse.core
+
+import grails.core.GrailsApplication
+import grails.gorm.transactions.Transactional
+import org.apache.commons.lang.WordUtils
+
+// TODO: Try to merge this logic into IdentifierService. We'd need to support a new ${abbreviation} keyword in the
+//       format that would get injected to based on the same logic that we have here. Then we'd add a new
+//       identifier.x.abbreviation.field = "path.to.field" config that we can use to extract the field from the entity.
+//       Or if that's too challenging, we can simply return the string to use via abstract method in IdentifierService.
+@Transactional
+class OrganizationIdentifierService {
+
+    GrailsApplication grailsApplication
+
+    /**
+     * Generates a unique id that resembles the given name.
+     *
+     * Ex: Given a name "Big Bad Guys" it might generate "BBG", or "BB0" if "BBG" already exists (up to "BB9").
+     * Ex: Given a name "Biggie" it might generate "BIG", or "BI0" if "BIG" already exists (up to "BB9").
+     */
+    String generate(String name) {
+        Integer minSize = grailsApplication.config.getProperty('openboxes.identifier.organization.minSize', Integer)
```

Comment:
Potential Tech Huddle conversation to be had ... do we provide the config properties as part of the Params / Context object passed to the generate method OR do we always inject these into the service using the @Value annotation.

Reply:
yeah we should discuss. I think @Value is the recommended approach in 99% of cases. There are situations here where we need to fetch individual settings by feature type, and for those it makes more sense to use the config service since we don't want to assume which feature settings are defined since they'll vary between implementations.

also the only reason it's using the old way here is because I didn't want to change the code until I had time to revisit and properly refactor it.